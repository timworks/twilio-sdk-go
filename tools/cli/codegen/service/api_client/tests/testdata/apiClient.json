{
  "name": "Service",
  "packageName": "service",
  "imports": [
    "github.com/timworks/twilio-sdk-go/client",
    "github.com/timworks/twilio-sdk-go/service/proxy/v1/service/phone_number",
    "github.com/timworks/twilio-sdk-go/service/proxy/v1/service/phone_numbers"
  ],
  "config": {
    "beta": true
  },
  "properties": [
    {
      "name": "sid",
      "type": "string"
    }
  ],
  "subClients": [
    {
      "name": "Phone Numbers",
      "packageName": "phone_numbers",
      "properties": [
        {
          "name": "serviceSid",
          "type": "string",
          "parentProperty": "sid"
        }
      ]
    },
    {
      "name": "Phone Number",
      "packageName": "phone_number",
      "functionParams": [
        {
          "name": "phoneNumberSid",
          "type": "string"
        }
      ],
      "properties": [
        {
          "name": "serviceSid",
          "type": "string",
          "parentProperty": "sid"
        },
        {
          "name": "sid",
          "type": "string",
          "functionParameter": "phoneNumberSid"
        }
      ]
    }
  ]
}
