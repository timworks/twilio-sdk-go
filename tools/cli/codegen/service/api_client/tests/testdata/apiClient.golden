// Package service contains auto-generated files. DO NOT MODIFY
package service

import "github.com/timworks/twilio-sdk-go/client"
import "github.com/timworks/twilio-sdk-go/service/proxy/v1/service/phone_number"
import "github.com/timworks/twilio-sdk-go/service/proxy/v1/service/phone_numbers"

// This client is currently in beta and subject to change. Please use with caution
type Client struct {
	client *client.Client

	sid string

	PhoneNumbers *phone_numbers.Client
	PhoneNumber  func(string) *phone_number.Client
}

// ClientProperties are the properties required to manage the service resources
type ClientProperties struct {
	Sid string
}

// New creates a new instance of the service client
func New(client *client.Client, properties ClientProperties) *Client {
	return &Client{
		client: client,

		sid: properties.Sid,

		PhoneNumbers: phone_numbers.New(client, phone_numbers.ClientProperties{
			ServiceSid: properties.Sid,
		}),
		PhoneNumber: func(phoneNumberSid string) *phone_number.Client {
			return phone_number.New(client, phone_number.ClientProperties{
				ServiceSid: properties.Sid,
				Sid:        phoneNumberSid,
			})
		},
	}
}
